### PR TITLE
chore(zod): update ssh key zod

### PR DESCRIPTION
**Pull request by maghirardelli**
Issue #, if available: GALI-2325

Description of changes:
Add zod conditions and fix unit/integ test for ssh key parsers

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts [hunk 1]
```diff
@@ -44,28 +43,52 @@ describe('Delete Key Pair negative tests', () => {
...
+      beforeEach(() => {
+        invalidSshKeyId = `sshkey-malformed`;
+      });
+
+      test.each(testBundle)('it throws 400', async (testCase) => {
+        const { username, session: sessionFunc, projectId: projectIdFunc } = testCase;
+        const session = sessionFunc();
+        const projectId = projectIdFunc();
+
+        console.log(`as ${username}`);
```
Comment (htztomic) [inline solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts:84]: do you need this console log?

Comment (maghirardelli) [reply solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts:84]: yup. it is for testing purposes for the logs to know which session it is running

File: workbench-core/base/src/utilities/textUtil.ts [hunk 2]
```diff
@@ -39,13 +39,15 @@ const awsAccountIdRegExpAsString: string = '^[0-9]{12}$';
...
 const provisionArtifactIdRegExpString: string = 'pa-[0-9a-zA-Z]{13}';
+const validSshKeyUuidRegExpAsString: string = '[0-9a-f]{64}';
 
 const envTypeIdRegExpString: string = `^${resourceTypeToKey.envType.toLowerCase()}-${productIdRegExpString},${provisionArtifactIdRegExpString}$`;
 const envTypeConfigIdRegExpString: string = `^${resourceTypeToKey.envTypeConfig.toLowerCase()}-${uuidRegExpAsString}$`;
 const projIdRegExpString: string = `^${resourceTypeToKey.project.toLowerCase()}-${uuidRegExpAsString}$`;
 const envIdRegExpString: string = `^${resourceTypeToKey.environment.toLowerCase()}-${uuidRegExpAsString}$`;
 const costCenterIdRegExpString: string = `^${resourceTypeToKey.costCenter.toLowerCase()}-${uuidRegExpAsString}$`;
 const accountIdRegExpString: string = `^${resourceTypeToKey.account.toLowerCase()}-${uuidRegExpAsString}$`;
+const sshKeyIdRegExpString: string = `${resourceTypeToKey.sshKey.toLowerCase()}-${validSshKeyUuidRegExpAsString}`;
```
Comment (janeyuaws) [inline workbench-core/base/src/utilities/textUtil.ts:50]: There should be unit test on this

File: workbench-core/base/src/utilities/validatorHelper.ts [hunk 3]
```diff
@@ -103,6 +105,10 @@ z.ZodString.prototype.accountId = function (): ZodString {
   return this.regex(accountIdRegex(), { message: invalidIdMessage });
 };
 
+z.ZodString.prototype.sshKeyId = function (): ZodString {
```
Comment (janeyuaws) [inline workbench-core/base/src/utilities/validatorHelper.ts:108]: There should be unit test on this

File: solutions/swb-reference/src/services/sshKeyService.test.ts [hunk 4]
```diff
@@ -53,8 +53,8 @@ describe('SshKeyService', () => {
     let listUserSshKeysForProjectRequest: ListUserSshKeysForProjectRequest;
     beforeEach(() => {
       listUserSshKeysForProjectRequest = {
-        projectId: 'proj-123',
-        userId: 'user-123'
+        projectId: 'proj-1234abcd-1111-abcd-1234-abcd1234abcd',
```
Comment (janeyuaws) [inline solutions/swb-reference/src/services/sshKeyService.test.ts:56]: I'd suggest you save mock values (mockProjectId, mockUserId) as constants at the top most describe()

File: solutions/swb-app/src/sshKeys/sendPublicKeyRequest.ts [hunk 5]
```diff
@@ -2,14 +2,14 @@
...
 
 // eslint-disable-next-line @rushstack/typedef-var
 export const SendPublicKeyRequestParser = z
   .object({
-    projectId: z.string(),
-    environmentId: z.string(),
-    userId: z.string()
+    projectId: z.string().projId().required(),
+    environmentId: z.string().envId().required(),
+    userId: z.string().required()
```
Comment (janeyuaws) [inline solutions/swb-app/src/sshKeys/sendPublicKeyRequest.ts:12]: nit: .userId().required()

File: solutions/swb-app/src/sshKeys/createSshKeyRequest.ts [hunk 6]
```diff
@@ -2,13 +2,13 @@
...
-import { z } from 'zod';
+import { z } from '@aws/workbench-core-base';
 
 // eslint-disable-next-line @rushstack/typedef-var
 export const CreateSshKeyRequestParser = z
   .object({
-    projectId: z.string(),
-    userId: z.string()
+    projectId: z.string().projId().required(),
+    userId: z.string().required()
```
Comment (htztomic) [inline solutions/swb-app/src/sshKeys/createSshKeyRequest.ts:11]: I'd recommend setting the entire object as required if you are requiring all values

File: solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts [hunk 7]
```diff
@@ -44,28 +43,52 @@ describe('Delete Key Pair negative tests', () => {
...
 
-    beforeEach(() => {
-      invalidSshKeyId = `sshkey-0000000000000000000000000000000000000000000000000000000000000000`;
+    describe('when key does not exist', () => {
+      let invalidSshKeyId: string;
+      beforeEach(() => {
+        invalidSshKeyId = `sshkey-0000000000000000000000000000000000000000000000000000000000000000`;
+      });
+
+      test.each(testBundle)('it throws 403', async (testCase) => {
```
Comment (janeyuaws) [inline solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts:52]: error code seems to be 404?

File: solutions/swb-app/src/sshKeys/createSshKeyRequest.ts (hunk 6 above)
Comment (janeyuaws) [inline solutions/swb-app/src/sshKeys/createSshKeyRequest.ts:11]: unfortunately there should be an interface in ZodString on userId(), and these work should have been covered in the generic parser ticket.....

File: workbench-core/base/src/utilities/textUtil.ts (hunk 2 above)
Comment (janeyuaws) [reply workbench-core/base/src/utilities/textUtil.ts:50]: nit: I'd suggest you put begin and ending regex ^$ here

File: solutions/swb-reference/src/services/sshKeyService.test.ts (hunk 4 above)
Comment (maghirardelli) [reply solutions/swb-reference/src/services/sshKeyService.test.ts:56]: Good point! Would have saved me some finding these values!

File: solutions/swb-app/src/sshKeys/createSshKeyRequest.ts (hunk 6 above)
Comment (maghirardelli) [reply solutions/swb-app/src/sshKeys/createSshKeyRequest.ts:11]: required is actually setting the length common requirements as well (ie. not just exist nor just nonempty). But it is a good point that there requests do need everything so the entire zod object should be required too!

File: solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts (hunk 7 above)
Comment (maghirardelli) [reply solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts:52]: oopsy thank you for catching

File: solutions/swb-app/src/sshKeys/createSshKeyRequest.ts (hunk 6 above)
Comment (maghirardelli) [reply solutions/swb-app/src/sshKeys/createSshKeyRequest.ts:11]: got it. I figured someone else had that ticket but I guess #scope-creep and I will add it here!

File: solutions/swb-app/src/projects/disassociateUserFromProjectRequest.ts [hunk 8]
```diff
@@ -3,12 +3,12 @@
...
 
-import { z } from 'zod';
+import { z } from '@aws/workbench-core-base';
 
 // eslint-disable-next-line @rushstack/typedef-var
 export const DisassociateUserFromProjectRequestParser = z
   .object({
-    userId: z.string(),
+    userId: z.string().userId().required(),
     projectId: z.string()
```
Comment (janeyuaws) [inline solutions/swb-app/src/projects/disassociateUserFromProjectRequest.ts:12]: nit: .projId().required(),

File: solutions/swb-app/src/sshKeys/createSshKeyResponse.ts [hunk 9]
```diff
@@ -3,16 +3,17 @@
...
 export const CreateSshKeyResponseParser = z
   .object({
-    projectId: z.string(),
-    privateKey: z.string(),
-    id: z.string(),
-    owner: z.string()
+    projectId: z.string().projId().required(),
+    privateKey: z.string().required(),
+    id: z.string().sshKeyId().required(),
+    owner: z.string().required()
```
Comment (janeyuaws) [inline solutions/swb-app/src/sshKeys/createSshKeyResponse.ts:14]: nit:  userId() parser

File: solutions/swb-app/src/sshKeys/sshKey.ts [hunk 10]
```diff
@@ -3,15 +3,18 @@
...
-  projectId: z.string(),
-  owner: z.string(),
-  publicKey: z.string(),
-  createTime: z.string()
-});
+export const SshKeyParser = z
+  .object({
+    sshKeyId: z.string().sshKeyId().required(),
+    projectId: z.string().projId().required(),
+    owner: z.string().required(),
```
Comment (janeyuaws) [inline solutions/swb-app/src/sshKeys/sshKey.ts:13]: same as above

File: solutions/swb-app/src/projects/disassociateUserFromProjectRequest.ts (hunk 8 above)
Comment (maghirardelli) [reply solutions/swb-app/src/projects/disassociateUserFromProjectRequest.ts:12]: for the record this is more scope creep but you are right to call it out that it needs to be fixed!

**Review by janeyuaws — APPROVED** [review]
great work!

**Pull request merged**